recipes/ignition-math/all/conanfile.py

```diff
@@ -104,46 +105,56 @@ def package(self):
-        self.cpp_info.components["libignition-math"].names["cmake_find_package_multi"] = "ignition-math{}".format(version_major)
-        self.cpp_info.components["libignition-math"].names["pkg_config"] = "ignition-math{}".format(version_major)
-
-        # FIXME: create in file ignition-math6-eigen3-config.cmake
-        self.cpp_info.components["libignition-math-eigen3"].libs = []
-        self.cpp_info.components["libignition-math-eigen3"].requires = ["libignition-math", "eigen::eigen"]
-        self.cpp_info.components["libignition-math-eigen3"].names["cmake_find_package"] = "ignition-math{}-eigen3".format(version_major)
-        self.cpp_info.components["libignition-math-eigen3"].names["cmake_find_package_multi"] = "ignition-math{}-eigen3".format(version_major)
-        self.cpp_info.components["libignition-math-eigen3"].names["pkg_config"] = "ignition-math{}-eigen3".format(version_major)
-
-        # FIXME: create in file ignition-math6-all-config.cmake
-        self.cpp_info.components["libignition-math-all"].libs = []
-        self.cpp_info.components["libignition-math-all"].requires = ["libignition-math-eigen3"]
-        self.cpp_info.components["libignition-math-all"].names["cmake_find_package"] = "ignition-math{}-all".format(version_major)
-        self.cpp_info.components["libignition-math-all"].names["cmake_find_package_multi"] = "ignition-math{}-all".format(version_major)
-        self.cpp_info.components["libignition-math-all"].requires.append("eigen::eigen")
-        self.cpp_info.components["libignition-math-all"].requires.append("doxygen::doxygen")
-        self.cpp_info.components["libignition-math-all"].requires.append("swig::swig")
-
-        self.env_info.LD_LIBRARY_PATH.extend([
-            os.path.join(self.package_folder, x) for x in self.cpp_info.libdirs
-        ])
-        self.env_info.PATH.extend([
-            os.path.join(self.package_folder, x) for x in self.cpp_info.bindirs
-        ])
+        lib_name = f"ignition-math{version_major}"
+        self.cpp_info.names["cmake_find_package"] = lib_name
+        self.cpp_info.names["cmake_find_package_multi"] = lib_name
+        self.cpp_info.names["cmake_paths"] = lib_name
+
+        self.cpp_info.components[lib_name].names["cmake_find_package"] = lib_name
+        self.cpp_info.components[lib_name].names["cmake_find_package_multi"] = lib_name
+        self.cpp_info.components[lib_name].names["cmake_paths"] = lib_name
+        self.cpp_info.components[lib_name].libs = [lib_name]
+        self.cpp_info.components[lib_name].includedirs.append(os.path.join("include", "ignition", "math"+version_major))
+        self.cpp_info.components[lib_name].requires = ["swig::swig", "eigen::eigen", "doxygen::doxygen"]
+        self.cpp_info.components[lib_name].build_modules["cmake_find_package"] = [self._module_file_rel_path]
+        self.cpp_info.components[lib_name].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
```

Comment:
I'm not 100% sure this suggestion will fix the hook warnings.
```suggestion
        self.cpp_info.components[lib_name].build_modules["cmake_find_package"] = [self._module_file_rel_path]
        self.cpp_info.components[lib_name].builddirs["cmake_find_package"] = [self._module_file_rel_path]
        self.cpp_info.components[lib_name].build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
        self.cpp_info.components[lib_name].builddirs["cmake_find_package_multi"] = [self._module_file_rel_path]
```

Reply:
thanks for the suggestion. I applied this manually.  waiting for the CI build

Reply:
@madebr your suggestions fixed the hooks warnings. Thanks :smile: